src/receivers/SocketModeReceiver.spec.ts

```diff
@@ -0,0 +1,288 @@
+      );
+      const SocketModeReceiver = await importSocketModeReceiver(overrides);
+
+      const receiver = new SocketModeReceiver({
+        appToken: 'my-secret',
+        logger: noopLogger,
+        clientId: 'my-clientId',
+        clientSecret: 'my-client-secret',
+        stateSecret: 'state-secret',
+        scopes: ['channels:read'],
+        installerOptions: {
+          authVersion: 'v2',
+          userScopes: ['chat:write'],
+        },
+      });
+      assert.isNotNull(receiver);
+      assert.isOk(fakeServer.listen.calledWith(3000));
+    });
+    it('should allow for customizing port the socket listens on', async () => {
+      // Arrange
+      const fakeServer = new FakeServer();
+      const fakeCreateServer = sinon.fake.returns(fakeServer);
+      const overrides = mergeOverrides(
+        withHttpCreateServer(fakeCreateServer),
+        withHttpsCreateServer(sinon.fake.throws('Should not be used.')),
+      );
+      const SocketModeReceiver = await importSocketModeReceiver(overrides);
+
+      const customPort = 1337;
+      const receiver = new SocketModeReceiver({
+        appToken: 'my-secret',
+        logger: noopLogger,
+        clientId: 'my-clientId',
+        clientSecret: 'my-client-secret',
+        stateSecret: 'state-secret',
+        scopes: ['channels:read'],
+        installerOptions: {
+          authVersion: 'v2',
+          userScopes: ['chat:write'],
+          port: customPort,
+        },
+      });
+      assert.isNotNull(receiver);
+      assert.isOk(fakeServer.listen.calledWith(customPort));
+    });
+    it('should invoke installer handleCallback if a request comes into the redirect URI path', async () => {
+      // Arrange
+      const fakeServer = new FakeServer();
+      let listener = (_req: any, _res: any) => {};
+      const fakeCreateServer = sinon.fake((handler: (req: any, res: any) => void) => {
+        listener = handler;
+        return fakeServer;
+      });
+      const installProviderStub = sinon.createStubInstance(InstallProvider);
+      const overrides = mergeOverrides(
+        withHttpCreateServer(fakeCreateServer),
+        withHttpsCreateServer(sinon.fake.throws('Should not be used.')),
+      );
+      const SocketModeReceiver = await importSocketModeReceiver(overrides);
+
+      const callbackOptions = {
+        failure: () => {},
+        success: () => {},
+      };
+      const receiver = new SocketModeReceiver({
+        appToken: 'my-secret',
+        logger: noopLogger,
+        clientId: 'my-clientId',
+        clientSecret: 'my-client-secret',
+        stateSecret: 'state-secret',
+        scopes: ['channels:read'],
+        installerOptions: {
+          authVersion: 'v2',
+          userScopes: ['chat:write'],
+          redirectUriPath: '/heyo',
+          callbackOptions,
+        },
+      });
+      assert.isNotNull(receiver);
+      receiver.installer = installProviderStub as unknown as InstallProvider;
```

Comment:
I must admit I don't have a ton of experience with TypeScript, and I feel shame for this `as unknown as whatever` type casting circus I've got going on here; if anyone knows of a cleaner way to do this, I would love to learn. FWIW, the typescript compiler told me to do this. So I blame the compiler 😇